lib/streams/request-stream.js

```diff
@@ -55,23 +48,14 @@ function RequestStream(requestStream, options) {
 			}
 		}
 	});
-	if(requestStream.method !== 'GET') {
-		inputPassThrough.pipe(requestStream, { noConvert: true });
+	CompoundDuplex.call(this, options);
+	if(requestStream.method === 'GET') {
+		this._compoundWrapNoWritablePipe = true;
```

Comment:
RequestStream is indeed a bit of a special case, since it can either be Duplex or Readable ... but ideally there'd be a less hacky way to deal with that.  Maybe automatically ending the writable portion?